lib/orbf/rules_engine/data/contract.rb

```diff
@@ -0,0 +1,52 @@
+module Orbf
+  module RulesEngine
+    class Contract
+      KNOWN_FIELDS = %w[startContract endContract id org_unit].freeze
+      attr_reader :id, :start_period, :end_period, :field_values
+      def initialize(field_values, calendar)
+        @field_values = field_values
+        @id = field_values["id"]
+        @org_unit = field_values["org_unit"]
+        @start_period = field_values["startContract"].gsub("-", "").slice(0, 6)
+        @end_period = field_values["endContract"].gsub("-", "").slice(0, 6)
+        @calendar = calendar
+      end
+
+      def match_period?(period)
+        start_month_period = @calendar.periods(period, "monthly")[0]
+        (
+          start_period <= start_month_period && start_month_period <= end_period
+        )
+      end
+
+      def org_unit_id
+        @org_unit["id"]
+      end
+
+      def org_unit_name
+        @org_unit["name"]
+      end
+
+      def codes
```

Comment:
That's to get all codes of groups (ie fields that are not the 3 "main ones") ?

Reply:
yes but "open to more data elements/optionsSets", the lowercase is to be compliant with the groups returned by the pyramid.